public/app/features/logs/components/LogRows.tsx

```diff
@@ -85,6 +94,42 @@ class UnThemedLogRows extends PureComponent<Props, State> {
     }
   };
 
+  handleSelection = (e: MouseEvent<HTMLTableRowElement>, row: LogRowModel): boolean => {
+    // Selection is ignored if the filter functions are not defined.
+    if (!this.props.onClickFilterOutValue && !this.props.onClickFilterValue) {
+      return false;
+    }
+    const selection = document.getSelection()?.toString();
+    if (!selection) {
+      return false;
+    }
+    this.setState({
+      selection,
+      menuCoordinates: { x: e.clientX, y: e.clientY },
+      selectedRow: row,
+    });
+    return true;
+  };
+
+  handleDeselection = (e: Event) => {
+    if (targetIsElement(e.target) && e.target?.getAttribute('role') === 'menuitem') {
+      // Delegate closing the menu to the popover component.
+      return;
+    }
+    if (document.getSelection()?.toString()) {
+      return;
+    }
```

Comment:
I was wondering if we could somehow scope that to the log rows? Because now, basically every selection will keep the menu open:


https://github.com/grafana/grafana/assets/8092184/6dc79f84-4d0c-43e1-bb96-8af57c05b7ae

Reply:
Now closing then the selection event comes from outside of the log rows:

https://github.com/grafana/grafana/assets/1069378/eb545e93-1323-4595-8672-6ae93adbc3f8